source/lambda/tests/botomock.py

```diff
@@ -0,0 +1,12 @@
+from unittest import mock
+import os
+
+
+def mock_boto_env_vars():
```

Comment:
Why not an autouse fixture? Then you don't have to call it every time

Reply:
I'm of two minds about it. Automatically taking care of it is nice in that you can't forget about it, but it's also hell to debug if you don't know there's extra magic happening behind the scenes. 

I had also been having trouble getting the magic to behave correctly due to certain boto elements being loaded statically on file import. I have fixed that in all currently affected test files, but this approach was originally born out of needing to be able to mock the env vars before certain files were imported, and now I generally like that it's clear exactly what is being mocked when.

Reply:
That import behavior is the thing I was fixing in ASR, it is frustrating. You can also move the import of the file in question to be within a function or test, so that it occurs after mocking happens.

This fixture is a pretty common pattern, so it's not like it's totally unexpected: https://www.learnaws.org/2020/12/01/test-aws-code/

One reason to use an autouse fixture is that when you forget to add that call, and you write your local tests, and you miss a stub, if your local profile has the permissions (mine is frequently on Admin) you get real calls into your account, which is not fun. Less catastrophically, it can also work locally but fail in the pipeline.